kernel/thread.c

```diff
@@ -119,13 +119,9 @@ void _impl_k_thread_custom_data_set(void *value)
 }
 
 #ifdef CONFIG_USERSPACE
-u32_t _handler_k_thread_custom_data_set(u32_t arg1, u32_t arg2, u32_t arg3,
-					u32_t arg4, u32_t arg5, u32_t arg6,
-					void *ssf)
+_SYSCALL_HANDLER1(k_thread_custom_data_set, data)
 {
-	_SYSCALL_ARG1;
-
-	_impl_k_thread_custom_data_set((void *)arg1);
+	_impl_k_thread_custom_data_set((void *)data);
```

Comment:
Could be a SIMPLE syscall

Reply:
no, data is not a kernel object but an arbitrary pointer